source/user_defined_output/user_defined_output_plugin_creator.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <memory>
+
+#include "nighthawk/user_defined_output/user_defined_output_plugin.h"
+
+#include "external/envoy/source/common/common/statusor.h"
+#include "external/envoy_api/envoy/config/core/v3/extension.pb.h"
+
+namespace Nighthawk {
+
+/**
+ * @brief Creates the User Defined Output Plugins for a set of configs.
+ *
+ * @param configs the TypedExtensionConfigs, used to identify the plugin factories and to configure
+ * each plugin created.
+ * @param worker_number which worker these plugins will be associated with.
```

Comment:
worker_id might be a better name than worker_number

Reply:
It's a reasonable point, but we call this worker_number throughout nighthawk, including in the main place for setting it: https://github.com/envoyproxy/nighthawk/blob/main/source/client/client_worker_impl.h#L29

Okay to keep it like this, given that?

Reply:
SGTM.